vochain/scrutinizer/vote.go

```diff
@@ -220,34 +220,21 @@ func (s *Scrutinizer) GetEnvelopeHeight(processID []byte) (uint64, error) {
 		// If no processID is provided, count all envelopes
 		return atomic.LoadUint64(&s.countTotalEnvelopes), nil
 	}
-
-	// The DB's count operation can error, so the cache entry stores either
-	// a result uint64 or an error.
-	// If an error occurs, right now it's "sticky", in the sense that it
-	// remains there until the entry is evicted as per the LRU rules.
-
-	val := s.envelopeHeightCache.GetAndUpdate(string(processID), func(prev interface{}) interface{} {
-		if prev != nil {
-			// If it's already in the cache, use it as-is.
-			return prev
-		}
-		// TODO: Warning, int can overflow
-		count, err := s.db.Count(&indexertypes.VoteReference{},
-			badgerhold.Where("ProcessID").Eq(processID).Index("ProcessID"))
-		if err != nil {
-			return err
-		}
-		return uint64(count)
-	})
-
-	switch val := val.(type) {
-	case error:
-		return 0, val
-	case uint64:
-		return val, nil
-	default:
-		panic(fmt.Sprintf("unexpected type: %T", val))
+	// Check if the envelope height is cached
+	val := s.envelopeHeightCache.Get(string(processID))
+	if val != nil {
+		return val.(uint64), nil
+	}
+	// If not cached, make the expensive query
+	results := &indexertypes.Results{}
+	if err := s.db.FindOne(results, badgerhold.Where(badgerhold.Key).Eq(processID)); err != nil {
+		return 0, err
```

Comment:
@p4u it doesn't make sense to me that we would only be able to fetch envelope height from a process that has results. For an encrypted process that hasn't completed, for example, will results be available to fetch the envelope height?

In either case I think we should check for ErrNotFound here and either manually count the envelopes, return 0, or return an error message that gives more context

Reply:
I don't follow you @NateWilliams2, if ErrNotFound a 0 and the error are returned

Are you saying that Encrypted processes do not have a Results entry? In that case yes, we should create also a results entry for Encrypted results and count the envelopes. But I think (need to check the code to be sure) that results are created since we need to count the Weight too.

Reply:
@p4u right, i guess what I'm describing has more to do with how this method is used. I brought this up because of the error on friday we had with getProcessSummary returning an ErrNotFound due to the results not being available. It seems like this should be the expected behavior though, so never mind :)